samples/sensor/sts4x/src/main.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Jan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+#if !DT_HAS_COMPAT_STATUS_OKAY(sensirion_sts4x)
+#error "No sensirion,sts4x compatible node found in the device tree"
+#endif
+
+int main(void)
+{
+	const struct device *const sts = DEVICE_DT_GET_ANY(sensirion_sts4x);
```

Comment:
I think this sample should be change from `sts4x/` to `amb_temp_polling/` and use `DT_ALIAS` since it is very generic.
Then all other simple temperature sensors can use this sample.